Source/Scene/Scene.js

```diff
@@ -3549,6 +3543,279 @@ define([
         }
 
         return result;
+    }
+
+    var scratchRight = new Cartesian3();
+    var scratchUp = new Cartesian3();
+
+    function pickFromRay(scene, ray, pickPosition, pickObject) {
```

Comment:
Is it cleaner to put the picking functions in a separate private file just so Scene.js isn't so large?

Reply:
OK.